set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/SearchUserSetRest.java

```diff
@@ -103,6 +119,38 @@ public UserSetQueryBuilder getQueryBuilder() {
 	}
     }
 
+    /**
+     * This method retrieves item ids from the closed userSet to build query e.g.
+     * https://api.europeana.eu/api/v2/search.json?profile=minimal&
+     * query=europeana_id%3A(%22%2F08641%2F1037479000000476635%22%20OR%20%20%22%2F08641%2F1037479000000476943%22)
+     * &rows=12&start=1
+     * @param userSet
+     * @return
+     */
+    private String buildItemIdsQueryFromClosedUserSet(UserSet userSet) {
+	// get item ids from the userSet
+        List<String> closedUserSetIds = userSet.getItems();
+	// use them to build the search query for retrieving item descriptions using minimal profile
+	// europeana_id is in format /collectionId/recordId, this can be easily extracted from the 
+        // full record ID by removing the base URL http://data.europeana.eu/item
+	// e.g. europeana_id:("/08641/1037479000000476635" OR  "/08641/1037479000000476943")
+	StringBuilder sb = new StringBuilder();
+        sb.append("https://api.europeana.eu/api/v2/search.json?profile=minimal&wskey=api2demo&query=europeana_id%3A(");
+        boolean firstPar = true;
+        for (String fullId : closedUserSetIds) {
+            String id = fullId.replace(WebUserSetFields.BASE_URL_DATA, "").replace("/", "%2F");
+    	    String elem = "%22" + id + "%22";
```

Comment:
you don't have to encode them here, simply call the url encoding after the query was built. 
Please use the native characters not the encoded ones in query building

Reply:
URL encoding called after query was build